### PR TITLE
Fix wrong type warning

### PR DESCRIPTION
**Pull request by rc-andres**
There are three commits in here for three different fixes:

1. We were building the aarch64 target against a kernel older than our minimum supported kernel. Unsure if this would case issues but we should at least be consistent. I have made the kernel versions match in the `Makefile` and fixes some of the imports that were making it not work when building in CI. 

2. Sometimes for unknown reasons we were pulling events out of `incomplete_events` with types that didn't match the `kretprobe`'s expected type. (i.e., the type would be a clone but the `kretprobe` would be for an `execve`). This is especially important because the incomplete events data was an `union` so we needed to know the right type to be able to touch the union correctly. To prevent this from ever happening this PR adds a map per type to guarantee that the `kretprobe` can always access the data without worrying that the type may be incorrect. That being said, because it is still unknown why the type was incorrect an extra check was added to make sure the `pid_tgid` inside the event is the same as the `pid_tgid` of the `kretprobe`. This in theory should never fail and if it did then it points to a race condition in the maps and we can look into that further. Also to be extra safe we are copying the event as soon as we load it so that we are guaranteed that our data won't change from under us when we are reading from it. 

3. `kretprobe`s are not guaranteed to fire so we cannot rely on it cleaning up any limited size map. Due to this I have changed the maps in process and network event programs to be lru maps so if they fill up they'll just evict older data. This is preferable to the map filling up and us unable to emit any new data.

### CODE REVIEW

**Review by pinealservo — COMMENTED** [review]
This looks like it will possibly resolve the confusion between clone and exec, but the individual syscall maps are probably way bigger than they need to be and also I suspect we could get everything we need from execve via the sched_process_exec tracepoint and avoid all the maps entirely.  It should have the post-interpreter-load args stored in `binprm` and point to the correct file for the actual executable.

File: src/process-events.c [hunk 1]
```diff
@@ -593,28 +628,28 @@ static __always_inline void enter_exec(struct pt_regs *ctx, const char __user *f
...
-    ret = bpf_map_update_elem(&incomplete_events, &pid_tgid, &event, BPF_ANY);
+    ret = bpf_map_update_elem(&incomplete_execs, &pid_tgid, &event, BPF_ANY);
     if (ret < 0)
     {
-        // should only happen if `incomplete_events` is filled
-        einfo.err = ret;
+        // should only happen if `incomplete_execs` is filled
+        pm->u.warning_info.info.err = ret;
         ret = -PMW_FILLED_EVENTS;
         goto Error;
```
[the remark this answers is not in the document]
Comment (rc-andres) [reply src/process-events.c:648]: true, I still want to emit an error if somehow this returns < 0 although it most likely wouldn't happen. There are just so many things that "shouldn't" happen that end up happening hence why I emit a warning on *anything* that technically has a error return code.

File: src/process-events.c [hunk 2]
```diff
@@ -85,12 +91,34 @@ struct bpf_map_def SEC("maps/process_events") process_events = {
...
+    .namespace = "",
+};
+
+// A map of unshares that have started (a kprobe) but are yet to finish
+// (the kretprobe).
+struct bpf_map_def SEC("maps/incomplete_unshare") incomplete_unshares = {
+    .type = BPF_MAP_TYPE_LRU_HASH,
+    .key_size = sizeof(u64),
+    .value_size = sizeof(incomplete_unshare_t),
+    .max_entries = 8 * 1024,
```
[the remark this answers is not in the document]
Comment (rc-andres) [reply src/process-events.c:111]: You are right, I'll reduce this to 256, I was just copying the number that was there for `incomplete_events`, but unshare is unlikely to ever be this high.

File: src/process-events.c (hunk 1 above)
[the remark this answers is not in the document]
Comment (rc-andres) [reply src/process-events.c]: I've renamed the warning to `PMW_UPDATE_MAP_ERROR`

**Comment by rc-andres** [comment]
> This looks like it will possibly resolve the confusion between clone and exec, but the individual syscall maps are probably way bigger than they need to be and also I suspect we could get everything we need from execve via the sched_process_exec tracepoint and avoid all the maps entirely. It should have the post-interpreter-load args stored in `binprm` and point to the correct file for the actual executable.

I think moving to a `tracepoint` is the right call but it may be too much of a lift for right now. I think we should focus on getting the data to be correct and without spamming so many warnings first. Then separately we should start looking into moving to a `tracepoint` for the `kretprobe` issues. I will do a separate PR that fixes the post-interpreter exec args/path that will also do away with the `exec_tids` and `incomplete_execs` since it should all be handled in the `kretprobe` but I didn't want to make this PR too large.

**Review by pinealservo — APPROVED** [review]
I'm fairly sure the changes will prevent the mixup of EXEC and CLONE events, but I am afraid we might end up combining mismatched start and exit sets in EXEC.  However, that's still an improvement and hopefully we'll be able to address the EXEC mixups more fully later.

**Pull request merged**